File: Number_Guessing_Game.cs
```CSharp
using System;

class Program
{
    static void Main(string[] args)
    {
        Random random = new Random();
        int secretNumber = random.Next(1, 101);
        int attempts = 0;
        bool isGuessed = false;

        Console.WriteLine("Welcome to the Number Guessing Game!");
        Console.WriteLine("Guess a number between 1 and 100:");

        while (!isGuessed)
        {
            string userInput = Console.ReadLine();
            int userGuess;

            if (int.TryParse(userInput, out userGuess))
            {
                attempts++;

                if (userGuess > secretNumber)
                {
                    Console.WriteLine("Too high! Try again:");
                }
                else if (userGuess < secretNumber)
                {
                    Console.WriteLine("Too low! Try again:");
                }
                else
                {
                    Console.WriteLine($"Congratulations! You've guessed the number in {attempts} attempts.");
                    isGuessed = true;
                }
            }
            else
            {
                Console.WriteLine("Invalid input! Please enter a number:");
            }
        }
    }
}
```

File: Number_Guessing_Game.py
import random

print("Welcome to the Number Guessing Game!")
print("Guess a number between 1 and 100.")

number_to_guess = random.randint(1, 100)
number_of_tries = 0

while True:
    user_guess = int(input("Your guess: "))
    number_of_tries += 1

    if user_guess < number_to_guess:
        print("Too low!")
    elif user_guess > number_to_guess:
        print("Too high!")
    else:
        print("Congratulations! You found the number in " + str(number_of_tries) + " tries.")
        break